### ISSUE TITLE
[Bug] Menu key does not work on mods list (accessibility)

### ISSUE THREAD

**Issue by Xetaxheb** [KSP-CKAN/CKAN#3445]
Background
----------
* Operating System: Windows 10
* CKAN Version: 1.30.4
* KSP Version: 1.12.2.3167

**Have you made any manual changes to your GameData folder (i.e., not via CKAN)?**
No

Problem
-------
**Describe the bug**
Pressing the "menu key" (between windows and right control) does not bring up "Labels/Reinstall/Download Contents/Purge Contents"

**Steps to reproduce**
Click on any random mod in list
Press "menu key" (https://en.wikipedia.org/wiki/Menu_key)

**Expected behavior**
The right-click menu pops up for highlighted mod after releasing menu key, navigable with arrows+enter (which it already is)

**Comment by HebaruSan** [comment]
Thanks for the report. #3446 should have a test build the Checks tab soon if you'd like to try it.

**Comment by Xetaxheb** [comment]
Yep works fine. One nuance that may warrant fixing if there's a simple solution: 
 - Usually the box is supposed to open on/next to the selected control (at the moment it's opening at cursor position, which is totally fine too; it's just better for clarity)

**Comment by HebaruSan** [comment]
Thanks for testing!

> * Usually the box is supposed to open on/next to the selected control (at the moment it's opening at cursor position, which is totally fine too; it's just better for clarity;  i would recommend the very center of "Name" field or the right edge of that same field at column border)

Yeah, I had it like that initially (and I noticed that Chrome does it that way), then I thought that if the user did want to navigate the menu with the mouse, it would be annoying to have to travel the mouse across the screen to do that. Then I tried pressing the menu key in the Atom editor and it opened at the cursor, so I thought I had found precedent for doing that.

> One other note: right clicking hyperlinks in metadata shows "copy link address" and doesn't function with menu key either, but enter opens it as a http/s handler anyway so it's still browser accessible and a non-issue

Good catch, thanks, we ought to be able to add this there, too.

**Comment by HebaruSan** [comment]
Any ideas for keyboard accessibility of the right click menu of the grid headers? There isn't really a way to focus the headers with the keyboard, and I don't know of an example of this in other applications.

**Comment by HebaruSan** [comment]
There's a new test build at the same place that should cover all of those additional context menus.

**Comment by Xetaxheb** [comment]
Finally had time to look at this (8a10141, 6 days ago), and the only things I can think of for accessibility left are: 

- adding a new menu at the top (file/settings/help) or maybe as a submenu under the settings menu: to enable/disable columns i.e. grid headers. just calling up the right-click menu itself would be fine.
- an additionally good thing would be a second submenu, 'Sort By ->" to likewise sort per column instead of needing to click the headers; could just call the same event as clicking them. hacky? maybe, but if it's stupid and it works, it's not stupid.
- making the launch/refresh/update/apply/filters/arrows accessible through tab cycling; or though a new menu at the top, with a separate menu for filters then
- Toggling installed/auto-installed/update boxes with spacebar is unclear because the dotted line "selection" box is missing; you can see it used to be there https://github.com/KSP-CKAN/CKAN/pull/2543 but I'm not seeing it anymore and I don't know if it's CKAN or my windows

**Comment by HebaruSan** [comment]
> * adding a new menu at the top (file/settings/help) or maybe as a submenu under the settings menu: to enable/disable columns i.e. grid headers. just calling up the right-click menu itself would be fine.
> * an additionally good thing would be a second submenu, 'Sort By ->" to likewise sort per column instead of needing to click the headers; could just call the same event as clicking them. hacky? maybe, but if it's stupid and it works, it's not stupid.

That makes sense! Sounds like the kind of thing I'd look for a in a "View" menu. It could have two submenus, _View &rarr; Columns_ with checkboxes and _View &rarr; Sort by_ with radio buttons, as you suggest. I might want to do that in a future pull request to keep this one from expanding too much.

> * making the launch/refresh/update/apply/filters/arrows accessible through tab cycling; or though a new menu at the top, with a separate menu for filters then

That I think we can do simply by setting `TabStop=true` for the toolbar control, will look into adding that.
